src/rust/src/x509/certificate.rs

```diff
@@ -319,6 +319,98 @@ impl Certificate {
             },
         )
     }
+
+    fn verify_signed_by<'p>(
+        &self,
+        py: pyo3::Python<'p>,
+        issuer: pyo3::PyRef<'_, Certificate>,
+    ) -> pyo3::PyResult<&'p pyo3::PyAny> {
+        let public_key = issuer.public_key(py)?;
+        let key_type = identify_public_key_type(py, public_key)?;
+        let hash_alg = self.signature_hash_algorithm(py)?;
+        match key_type {
+            sign::KeyType::Ed25519 | sign::KeyType::Ed448 => public_key.call_method1(
+                "verify",
+                (self.signature(py), self.tbs_certificate_bytes(py)?),
+            ),
+            sign::KeyType::Ec => {
+                let ec_mod = py.import("cryptography.hazmat.primitives.asymmetric.ec")?;
+                let ecdsa = ec_mod
+                    .getattr(crate::intern!(py, "ECDSA"))?
+                    .call1((hash_alg,))?;
+                public_key.call_method1(
+                    "verify",
+                    (self.signature(py), self.tbs_certificate_bytes(py)?, ecdsa),
+                )
+            }
+            sign::KeyType::Rsa => {
+                let padding_mod = py.import("cryptography.hazmat.primitives.asymmetric.padding")?;
+                let pkcs1v15 = padding_mod
+                    .getattr(crate::intern!(py, "PKCS1v15"))?
+                    .call0()?;
+                public_key.call_method1(
+                    "verify",
+                    (
+                        self.signature(py),
+                        self.tbs_certificate_bytes(py)?,
+                        pkcs1v15,
+                        hash_alg,
+                    ),
+                )
+            }
+            sign::KeyType::Dsa => public_key.call_method1(
+                "verify",
+                (
+                    self.signature(py),
+                    self.tbs_certificate_bytes(py)?,
+                    hash_alg,
+                ),
+            ),
+        }?;
```

Comment:
Put the `?` on each of the individual `call_method` calls, much easier to read

Reply:
I originally did that, but ran into rust coverage problems. Of course, this form didn't fix them, just reduced it to one line of incorrect missed coverage.

Reply:
that coverage issue would indicate missing failed sigs on one of the key types... which we shoudl test